src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -627,11 +688,19 @@ internal void ChangeRootCategoryExpandState(string categoryName, bool isExpanded
         internal void SearchAndUpdateResults()
         {
             if (!String.IsNullOrEmpty(SearchText.Trim()))
+            {
                 SearchAndUpdateResults(SearchText);
+            }
+            else // Search text is empty, clear search results.
+            {
```

Comment:
You can put  searchResults.Clear(); in SearchAndUpdateResults. there is a null check there.